src/kademlia/routing_table.cpp

```diff
@@ -558,6 +580,10 @@ routing_table::add_node_status_t routing_table::add_node_impl(node_entry e)
 	INVARIANT_CHECK;
 #endif
 
+	// don't add if the address isn't the right type
+	if (!native_address(e.addr()))
```

Comment:
We can get here via node_seen() when a node issues a bootstrap using the non-native IP version.